src/vs/code/electron-main/main.ts

```diff
@@ -174,6 +175,12 @@ function main(accessor: ServicesAccessor, mainIpcServer: Server, userEnv: platfo
 		// TODO@Joao: unfold this
 		windowsMainService = accessor.get(IWindowsMainService);
 
+		windowsMainService.onWindowClose(() => {
+			if (windowsMainService.getWindowCount() === 0) {
```

Comment:
@joaomoreno as I said before, this should have a check for Windows/Linux, because only there we close the app when closing the last window!